asio/include/xtt/asio/server_context.inl

```diff
@@ -0,0 +1,403 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <xtt/asio/error_category.hpp>
+
+namespace xtt {
+namespace asio {
+
+    template <typename GPKLookupCallback,
+              typename AssignIdCallback,
+              typename Handler>
+    void
+    server_context::async_do_read(server_handshake_context::io_buffer io_buf,
+                                  GPKLookupCallback async_lookup_gpk_callback,
```

Comment:
Maybe just `async_lookup_gpk` instead of `async_lookup_gpk_callback`.  Again to keep the parameter lists and usage sites more readable.